### PR TITLE
additional connection stats on advanced settings tab

### LINKED ISSUE #6568: Show connection stats inside the app
### Description
In order to debug some of the connection issues, it would be nice to enable connection stats logging on the client side. E.g., it can be hidden under a Profile->Advanced menu, and would display
- bytes/packets total counts sent in both directions
- current bandwidth usage
- average timeout values
- etc.

Related to #6567

### PR DESCRIPTION
**Pull request by rcullito**
Fixes #6568 

PR to display additional information useful in debugging around packets, bandwidth usage, and timeouts. 

Screenshot included below.

### CODE REVIEW

**Comment by chadyj** [comment]
Would be great to include screenshots in your PR's :)

I downloaded this build but didn't see anything new?

**Comment by rcullito** [comment]
Cheers @chadyj , yes that makes sense and sorry for the anti-climactic download 👀  I wanted to get something up to kick off discussion about running status-react against a local status-go (https://github.com/status-im/status.im/pull/176). I've updated the PR description but in the future if you'd like to settle on a label such as *no functional change* or something to better communicate this intention of early stage PRs, let me know 👍

**Comment by rcullito** [comment]
connectionstatisticslowerright

@chadyj  here is a first crack at displaying a subset of connection metrics. seems like future iterations would be in sync with design but for now I at least wanted to get the raw numbers out there. 

@adambabik I'm not sure if it is expected that values returned from the json-rpc call here would be 0 in a development environment? I had joined a few public chats with traffic, but, at least at time of query, didn't see any increased values in the debug metrics response. This is somewhat related to @mandrigin 's comment on https://github.com/status-im/status-react/issues/6568 about currently running services on the desktop node. I wasn't sure if the mere presence of a key in the response was enough to indicate that the given service was running? Or if a positive had to be found in order to conclude this? Thanks!

**Comment by adambabik** [comment]
@rcullito it is if you did not use this status-go branch https://github.com/status-im/status-go/tree/enable-metrics-by-default.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/status-im/status-go">status-im/status-go</a></strong></div><div>The Status module that consumes go-ethereum. Contribute to status-im/status-go development by creating an account on GitHub.</div></blockquote>

**Review by vkjr — APPROVED** [review]

**Review by maxhora — APPROVED** [review]

File: src/status_im/ui/components/desktop/events.cljs [hunk 1]
```diff
@@ -38,3 +43,37 @@
...
    (show-desktop-tab cofx tab-name)))
+
+(defn packet-keys [direction]
+  "helper for accessing deeply nested metrics keys"
+  {:pre [(#{:in :out} direction)]}
+  [:misc direction :packets :Overall])
+
+(handlers/register-handler-fx
+ :debug-metrics-success
+ (fn [{:keys [db]} [_ {:keys [p2p mailserver les whisper], :as metrics}]]
```
Comment (vitvly) [inline src/status_im/ui/components/desktop/events.cljs:42]: Extra comma, just jumped on me :)

**Review by vitvly — APPROVED** [review]

**Comment by rcullito** [comment]
@adambabik thanks for the link to the branch. and sorry yea saw you had mentioned that on the original issue for this. In terms of merging this to develop on the status-react side I suppose it makes sense to only coordinate that until after *enable-metrics-by-default* has been merged into develop on the status-go side. So I'll watch out for that assuming that's the case?

File: src/status_im/ui/components/desktop/events.cljs (hunk 1 above)
Comment (rcullito) [reply src/status_im/ui/components/desktop/events.cljs:42]: ah, nice catch @siphiuel thanks. should it be removed you're saying?

**Comment by adambabik** [comment]
> In terms of merging this to develop on the status-react side I suppose it makes sense to only coordinate that until after enable-metrics-by-default has been merged into develop on the status-go side. So I'll watch out for that assuming that's the case?

Yeah, if you are ready with this PR then we can speed up merging it on status-go side. The only thing that I would like to check on status-go side is compare memory usage. Collecting metrics might affect memory consumption.

**Comment by rcullito** [comment]
k, yea makes sense! guessing there will probably be later passes from our side for fancier data viz and such but for now it's ready to at least display the raw numbers until we find out what's most important for users to understand 

sounds good re: memory usage investigation. whenever it lands under a version, I'll just drop in a new status_go_version and then we can all test E2E :tada:

**Comment by adambabik** [comment]
This build should do it: https://ci.status.im/job/status-go/job/parallel/174/. I guess a version `181209-170036-21089d` can be used. 

After we check the memory consumption and confirm it's fine, I will merge a status-go PR.

**Comment by churik** [comment]
@siphiuel can you please rebase it and I'll try to rebuild?

**Comment by vitvly** [comment]
@churik rebased

**Comment by asemiankevich** [comment]
So if i got it correct, we want to have this on Desktop:

image

I am wondering why the values here are always 0? @adambabik @siphiuel any thoughts? I have joined some chats (public and 1x1), i still see Fetching messages meaning i am receiving some data. I am still digging into details but at the first glance all zeroes here are confusing and not informative.
Do you know any golden source to compare the values with the ones that actually `should be shown` there? Where i can check that?

**Comment by vitvly** [comment]
@asemiankevich i'll check what's up there, maybe there we're not using correct status-go version.

**Comment by asemiankevich** [comment]
> @asemiankevich i'll check what's up there, maybe there we're not using correct status-go version.

okay thanks!

meanwhile i checked that upgrade is not broken (from nightly to this PR). Also mobile tests look good. will check more, let me know the results of our research please

**Comment by vitvly** [comment]
@adambabik i've tried the branch locally with STATUS_GO_VERSION=enable-metrics-by-default, but the metrics returned are still 0s. Maybe there is something missing in the configuration?

```
{:rendezvous {:InboundTraffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :OutboundTraffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :txpool {:invalid {:Overall 0}, :pending {:discard {:Overall 0}, :nofunds {:Overall 0}, :ratelimit {:Overall 0}, :replace {:Overall 0}}, :queued {:discard {:Overall 0}, :nofunds {:Overall 0}, :ratelimit {:Overall 0}, :replace {:Overall 0}}, :underpriced {:Overall 0}}, :db {:preimage {:hits {:Overall 0}, :total {:Overall 0}}}, :mailserver {:requestValidationErrors {:Overall 0}, :archivedEnvelopes {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :archivedEnvelopesSize {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :sentEnvelopes {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :requestProcessNetTime {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}, :requests {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :syncRequests {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :requestsBatched {:Overall 0}, :processRequestErrors {:Overall 0}, :requestProcessTime {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}, :historicResponseErrors {:Overall 0}, :sentEnvelopesSize {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :requestErrors {:Overall 0}, :archiveErrors {:Overall 0}}, :trie {:cachemiss {:Overall 0}, :cacheunload {:Overall 0}, :memcache {:commit {:nodes {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :size {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :time {:Mean 0, :Measurements 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}}, :flush {:nodes {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :size {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :time {:Mean 0, :Measurements 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}}, :gc {:nodes {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :size {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :time {:Mean 0, :Measurements 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}}}}, :les {:misc {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}}, :discv5 {:InboundTraffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :OutboundTraffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :p2p {:InboundConnects {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :InboundTraffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :OutboundConnects {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :OutboundTraffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :chain {:inserts {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}}, :eth {:downloader {:bodies {:drop {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :req {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}, :timeout {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :headers {:drop {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :req {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}, :timeout {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :receipts {:drop {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :req {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}, :timeout {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :states {:drop {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :fetcher {:fetch {:bodies {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :headers {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :filter {:bodies {:in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :out {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :headers {:in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :out {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :prop {:announces {:dos {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :drop {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :out {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}}, :broadcasts {:dos {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :drop {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :in {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :out {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0, :Percentiles {:5 0, :20 0, :50 0, :80 0, :95 0}}}}}, :misc {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :prop {:blocks {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :hashes {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :txns {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}}, :req {:bodies {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :headers {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :receipts {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}, :states {:in {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}, :out {:packets {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}, :traffic {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}}}, :whisper {:envelopeErrFromFuture {:Overall 0}, :envelopeErrOversized {:Overall 0}, :envelopeAdded {:Overall 0}, :envelopeCleared {:Overall 0}, :envelopeErrLowPow {:Overall 0}, :envelopeErrVeryOld {:Overall 0}, :envelopeNewAdded {:Overall 0}, :envelopeErrNoBloomMatch {:Overall 0}, :envelopeErrExpired {:Overall 0}, :envelopeSize {:AvgRate01Min 0, :AvgRate05Min 0, :AvgRate15Min 0, :MeanRate 0, :Overall 0}}}
```

**Comment by adambabik** [comment]
@siphiuel do you have access to `geth.log`? There should be a log `Enabling metrics collection` if metrics are enabled. There is not other configuration.

But I think I might have found a problem. status-go for Desktop is built using `make statusgo-library`? If so, then build flags were not passed. I pushed [a commit](https://github.com/status-im/status-go/commit/a0ee44cce366a2367e33fb5593a339fe5dd77634) fixing that.

**Comment by vitvly** [comment]
Thank you @adambabik . After also adding a Desktop bridge for `verify()`, it works and displays at least some numbers.

**Comment by churik** [comment]
@siphiuel can you please rebase it to current develop branch (fix for linux was added)?

**Comment by vitvly** [comment]
@churik done

**Comment by churik** [comment]
@siphiuel mobile builds are failing with following errors:
IOS:
```
14:44:21 unzip:  cannot find or open modules/react-native-status/ios/RCTStatus/status-go-ios-enable-metrics-by-default.zip, modules/react-native-status/ios/RCTStatus/status-go-ios-enable-metrics-by-default.zip.zip or modules/react-native-status/ios/RCTStatus/status-go-ios-enable-metrics-by-default.zip.ZIP.
```
Android:
```
14:51:26 FAILURE: Build failed with an exception.
14:51:26 
14:51:26 * What went wrong:
14:51:26 Could not resolve all files for configuration ':app:releaseRuntimeClasspath'.
14:51:26 > Could not find status-im:status-go:enable-metrics-by-default.
```

Can you help with that?

**Comment by vitvly** [comment]
@churik yes, it's because the `enable-metrics-by-default` branch is not merged yet, and there is no proper ios/android build for it (desktop builds fetch the code and run the build themselves, so they are successful).
@adambabik - can you please estimate when the branch will be merged and we can use the functionality from a tagged status-go release?

**Comment by churik** [comment]
Tested on desktop builds only.
Depends on `enable-metrics-by-default` branch - since it will be merged in status-go - 0 will be shown in stats.
Also fixed `verify()` error on log in.

**Comment by vitvly** [comment]
Merged, closing manually.

**Comment by adambabik** [comment]
@siphiuel I will make sure it will be merged this week.

**Pull request closed without merging**